deploy/buildtools/buildconfig.py

```diff
@@ -1,177 +1,151 @@
-""" This converts the build configuration files into something usable by the
-manager """
-
 from time import strftime, gmtime
-import configparser
 import pprint
+from importlib import import_module
 
-from runtools.runtime_config import RuntimeHWDB
 from awstools.awstools import *
+from buildtools.buildfarmhostdispatcher import BuildFarmHostDispatcher
+
+# typing imports
+from typing import Set, Type, Any, Optional, Dict, TYPE_CHECKING
+# needed to avoid type-hint circular dependencies
+# TODO: Solved in 3.7.+ by "from __future__ import annotations" (see https://stackoverflow.com/questions/33837918/type-hints-solve-circular-dependency)
+#       and normal "import <module> as ..." syntax (see https://www.reddit.com/r/Python/comments/cug90e/how_to_not_create_circular_dependencies_when/)
```

Comment:
We need to bump to 3.7+ so that we can upgrade moto and enable the tests I added recently to `main`.  I think we should do it in a follow-on PR.  Now that the list of reasons is growing, I opened #968 to track the reasons why we want to bump.